backend/packages/Upgrade/src/api/controllers/StratificationController.ts

```diff
@@ -0,0 +1,240 @@
+   *                $ref: '#/definitions/FactorStrata'
+   *          '401':
+   *            description: Authorization Required Error
+   */
+  @Get()
+  public async getAllStratification(@Req() request: AppRequest): Promise<FactorStrata[]> {
+    return this.stratificatonService.getAllStratification(request.logger);
+  }
+
+  /**
+   * @swagger
+   * /stratification/download/{factor}:
+   *    get:
+   *      description: Get stratificationFactor CSV by factorName
+   *      tags:
+   *        - Stratification
+   *      produces:
+   *        - application/json
+   *      parameters:
+   *        - in: path
+   *          name: factor
+   *          description: Factor name
+   *          required: true
+   *          schema:
+   *            type: string
+   *      responses:
+   *        '200':
+   *          description: Get stratificationFactor detailed CSV by name
+   *        '401':
+   *          description: Authorization Required Error
+   *        '404':
+   *          description: Factor name not found
+   *        '500':
+   *          description: Internal Server Error, FactorName is not valid
+   */
+  @Get('/download/:factor')
+  public async getStratificationByFactor(
+    @Param('factor') factor: string,
+    @Req() request: AppRequest,
+    @Res() res: express.Response
+  ): Promise<any> {
+    if (!factor) {
+      return Promise.reject(new Error(SERVER_ERROR.MISSING_PARAMS + ' : stratification Factor should not be null.'));
+    }
+
+    const data = await this.stratificatonService.getCSVDataByFactor(factor, request.logger);
+
+    // Convert JSON data to CSV
+    const parser = new Parser();
+    const csv = parser.parse(data);
+
+    // return csv file with appropriate headers to request;
+    res.setHeader('Content-Type', 'text/csv; charset=UTF-8');
+    res.setHeader('Content-Disposition', 'attachment; filename="data.csv"');
+    return res.send(csv);
+  }
+
+  /**
+   * @swagger
+   * /stratification:
+   *    post:
+   *      description: Create a new stratificationFactor by CSV
+   *      tags:
+   *        - Stratification
+   *      produces:
+   *        - application/json
+   *      parameters:
+   *        - in: body
+   *          name: file
+   *          description: CSV file
+   *          required: true
+   *          schema:
+   *            type: file
+   *      responses:
+   *        '200':
+   *          description: Create a new UserStratificationFactors by CSV
+   *          schema:
+   *            type: array
+   *            items:
+   *              $ref: '#/definitions/UserStratificationFactor'
+   *        '401':
+   *          description: Authorization Required Error
+   *        '500':
+   *          description: Internal Server Error, Insert Error in database, CSV file is not valid
+   */
+  @Post()
+  @UseBefore(upload.single('file'))
+  public insertStratification(@Req() request: AppRequest): Promise<UserStratificationFactor[]> {
+    const csvData = request.body[0].file;
```

Comment:
Parsing logic should be moved inside the service. Controller should only pass the data to the service.